[FIX] shopfloor: single pack transfer, handle backorders

sebalix opened the pull request:
When processing a package with the single pack transfer scenario which
belongs to a transfer having several other packages, the remaining ones
are not put in a backorder and as such can not be processed anymore.

This commit ensure to put the processed package/package level in its own
move which will be validated in its own transfer, keeping the remaining
packages to process in the current transfer.

Ref. 2400

sebalix committed:
[FIX] shopfloor: single pack transfer, handle backorders

When processing a package with the single pack transfer scenario which
belongs to a transfer having several other packages, the remaining ones
are not put in a backorder and as such can not be processed anymore.

This commit ensure to put the processed package/package level in its own
move which will be validated in its own transfer, keeping the remaining
packages to process in the current transfer.

simahawk commented on shopfloor/services/single_pack_transfer.py:
```diff
@@ -256,7 +256,14 @@ def _set_destination_and_done(self, package_level, scanned_location):
         package_level.location_dest_id = scanned_location
         stock = self._actions_for("stock")
-        stock.validate_moves(package_level.move_line_ids.move_id)
+        package_moves = package_level.move_line_ids.move_id
```
@sebalix am I wrong or we have done this N times in many places for the same reason?
Can't we move this to `stock` action?

sebalix replied:
Hum yes, it has been done like this at least in the location content transfer. I'll try to move it to the stock action.

sebalix replied:
I added a commit adding a new method `put_package_level_in_move` in the `stock` action, used by the single pack transfer and location content transfer scenarios.

sebalix committed:
[IMP] shopfloor: new stock helper 'put_package_level_in_move'

New method 'put_package_level_in_move' to put a package in its own move,
making it easy to process afterward, without impacting other package
levels.

simahawk commented:
/ocabot merge patch
